### PR TITLE
support Intel performance counters on Windows

### PR DESCRIPTION
**Pull request by Trass3r**

### CODE REVIEW

File: renderdoc/driver/ihv/intel/intel_gl_counters.cpp [hunk 1]
```diff
@@ -116,8 +117,10 @@ void IntelGlCounters::addQuery(GLuint queryId)
   GL.glGetIntegerv(eGL_PERFQUERY_QUERY_NAME_LENGTH_MAX_INTEL, &len);
   query.name.resize(len);
   GLuint nCounters = 0;
+  GLuint nInstances = 0;
+  GLuint capsMask = 0;
   GL.glGetPerfQueryInfoINTEL(queryId, (GLuint)query.name.size(), &query.name[0], &query.size,
-                             &nCounters, NULL, NULL);
+                             &nCounters, &nInstances, &capsMask);
```
Comment (Trass3r) [inline renderdoc/driver/ihv/intel/intel_gl_counters.cpp:123]: They didn't work before because you must not pass null here.

**Comment by baldurk** [comment]
I've not forgotten about this PR, I just need to find time to test it. When the GL counters were initially contributed they were enabled for windows and linux, but the extension seemed to be broken on windows and would cause renderdoc to crash/hang when loading captures.

**Comment by Trass3r** [comment]
See my inline comment. It was incorrect API usage though iirc that part wasn't properly documented either.

**Comment by baldurk** [comment]
Sorry I don't see any inline comment, where is that?

**Comment by Trass3r** [comment]
Sorry it didn't get published.

**Comment by baldurk** [comment]
Ah OK I see. Yes I don't see anything that documents that behaviour in the extension, but it would make sense why there were errors.

I'll try to check this soon, @djdeath it would be good if you could verify that this will be fine for mesa (I assume it will be since the code doesn't care about those values).

**Comment by djdeath** [comment]
Can't see anything that would break mesa.

**Comment by baldurk** [comment]
For me it doesn't produce any interesting counters, only clones of the generic pipeline statistics ones (vertex counts, shader invocation counts, etc). It seems to work fine though and maybe that's just a limitation of my hardware or driver.

**Pull request merged**